packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverPageCount/TeachingPopoverPageCount.types.ts

```diff
@@ -1,17 +1,49 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type TeachingPopoverPageCountSlots = {
-  root: Slot<'div'>;
+  /**
+   * The element wrapping the carousel pagination. By default this is a div,
+   * it may contain icons or text depending on TeachingPopoverPageCountStyle
+   */
+  root: NonNullable<Slot<'div'>>;
+
+  /**
+   * The element for each carousel page in 'icon' mode.
+   */
+  carouselIcon: NonNullable<Slot<'button'>>;
+
+  /**
+   * The element for each carousel page in 'icon' mode.
+   */
+  carouselSelectedIcon: NonNullable<Slot<'button'>>;
+};
+
+export type TeachingPopoverPageCountStyle = 'text' | 'icon';
+
+export type TeachingPopoverPageCountState = ComponentState<TeachingPopoverPageCountSlots> & {
+  /**
+   * The current carousel page.
+   */
+  currentPage: number;
+  /**
+   * The total number of carousel pages, controlled by children within carousel itself.
+   */
+  totalPages: number;
+  /**
+   * Allows the carousel pagination to update the current page based on icon interaction.
+   */
+  setCurrentPage: (index: number) => void;
+  /**
+   * The style in which the page count is rendered.
+   */
+  countStyle: TeachingPopoverPageCountStyle;
 };
 
-/**
- * TeachingPopoverPageCount Props
- */
-export type TeachingPopoverPageCountProps = ComponentProps<TeachingPopoverPageCountSlots> & {};
+// For localization or customization, users may want to modify this for their own purposes
+export type TeachingPopoverPageCountChildRenderFunction = (currentPage: number, totalPages: number) => React.ReactNode;
```

Comment:
I don't think that render props is a typical design for Fluent and it's probably the first use case of this. Let's log an item to discuss this approach

Reply:
Resolution tracked by #30127.